narwhal/primary/src/primary.rs

```diff
@@ -268,6 +271,10 @@ impl Primary {
                 inbound_network_metrics,
             )))
             .layer(CallbackLayer::new(FailpointsMakeCallbackHandler::new()))
+            .layer(SetRequestHeaderLayer::overriding(
```

Comment:
Should this be on the `service` or on the `outbound_layer` instead? We want to add the `epoch` header on the outbound requests right?

Reply:
good call, moved to outbound

Reply:
So is this still needed here or was that meant to be a `SetResponseHeaderLayer` to address what @bmwill suggested?